src/compas/files/gltf/extensions.py

```diff
@@ -0,0 +1,335 @@
+        self.specular_texture = specular_texture
+        self.specular_color_factor = specular_color_factor
+        self.specular_color_texture = specular_color_texture
+
+    def to_data(self, texture_index_by_key, **kwargs):
+        dct = {}
+        if self.specular_factor is not None:
+            dct["specularFactor"] = self.specular_factor
+        if self.specular_texture is not None:
+            dct["specularTexture"] = self.specular_texture.to_data(texture_index_by_key)
+        if self.specular_color_factor is not None:
+            dct["specularColorFactor"] = self.specular_color_factor
+        if self.specular_color_texture is not None:
+            dct["specularColorTexture"] = self.specular_color_texture.to_data(
+                texture_index_by_key
+            )
+        if self.extras is not None:
+            dct["extras"] = self.extras
+        if self.extensions is not None:
+            dct["extensions"] = self.extensions_to_data()
+        return dct
+
+    @classmethod
+    def from_data(cls, dct):
+        if dct is None:
+            return None
+        return cls(
+            specular_factor=dct.get("specularFactor"),
+            specular_texture=TextureInfoData.from_data(dct.get("specularTexture")),
+            specular_color_factor=dct.get("specularColorFactor"),
+            specular_color_texture=TextureInfoData.from_data(
+                dct.get("specularColorTexture")
+            ),
+            extensions=cls.extensions_from_data(dct.get("extensions")),
+            extras=dct.get("extras"),
+        )
+
+
+class KHR_materials_ior(BaseGLTFDataClass):
+    """glTF extension that defines the optical transmission of a material.
+
+    https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_materials_ior
+    """
+
+    key = "KHR_materials_ior"
+
+    def __init__(
+        self,
+        ior=None,
+        extensions=None,
+        extras=None,
+    ):
+        super(KHR_materials_ior, self).__init__(extras, extensions)
+        self.ior = ior
+
+    def to_data(self, texture_index_by_key, **kwargs):
+        dct = {}
+        if self.ior is not None:
+            dct["ior"] = self.ior
+        if self.extras is not None:
+            dct["extras"] = self.extras
+        if self.extensions is not None:
+            dct["extensions"] = self.extensions_to_data()
+        return dct
+
+    @classmethod
+    def from_data(cls, dct):
+        if dct is None:
+            return None
+        return cls(
+            ior=dct.get("ior"),
+            extensions=cls.extensions_from_data(dct.get("extensions")),
+            extras=dct.get("extras"),
+        )
+
+
+class KHR_materials_clearcoat(BaseGLTFDataClass):
+    """glTF extension that defines the clearcoat material layer.
+
+    https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Khronos/KHR_materials_clearcoat
+    """
+
+    key = "KHR_materials_clearcoat"
+
+    def __init__(
+        self,
+        clearcoat_factor=None,
```

Comment:
there is probably a better way to handle this -you are right-, but then I would need to change the `gltf.data_classes` as well, since the same checking is performed there...

Reply:
I can't speak to the optionality of these specific attributes, but in general, many attributes in glTF are optional.  I believe though that it's invalid to include `null` values in the JSON part of a glTF.  When the values in question are `None`, those keys should not be added to the data which is then dumped to the JSON even if it is the value the user passed.  I agree though that there probably is a better way to do this.